### PR TITLE
ISPN-14966 Connected clients page

### PR DESCRIPTION
**Pull request by dpanshug**
https://issues.redhat.com/browse/ISPN-14966

### CODE REVIEW

File: src/app/utils/formatAge.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Formats a given date string into a friendly age format.
+ * The friendly age format includes hours, minutes, and seconds.
+ * @param dateString The date string to format.
+ * @returns The formatted age string.
+ */
+export function formatAge(dateString: string): string {
```
Comment (karesti) [inline src/app/utils/formatAge.ts:7]: there is a library already included that does this formatting I think, no need to do it by coding a function again, but I'm not sure.

Comment (karesti) [reply src/app/utils/formatAge.ts:7]: if you add functions like this, you need a unit test to cover as well

File: src/services/serverService.ts [hunk 2]
```diff
@@ -28,4 +28,11 @@ export class ServerService {
   public async getLatestVersion(): Promise<Either<ActionResponse, string>> {
```
Comment (karesti) [inline src/services/serverService.ts:28]: Unrelated, but I spotted this

The version we are using in the Server should not be grabbed from the github API
There is an endpoint (about uses it) that displays the version of the server we are using.
Is misleading, IMO, putting this version like current version as the latest one, and won't work for DG
IMO this call should be removed and use the server version API

**Review by karesti — COMMENTED** [review]
Screenshot 2023-07-03 at 18 36 08

The endpoint is unavailable still as I see (running dev01)
 I will review the behaviour from main when these comments will be addressed

**Comment by karesti** [comment]
Screenshot 2023-07-04 at 09 34 51

**Comment by karesti** [comment]
- Display principal N/A, never display "null"
- connection names are not accurate, this should be fixed not tu duplicate names
- if client version is null, display N/A 
- Principal tab should be bigger

@tristantarrant this is what I'm getting running the console
-

File: src/services/serverService.ts [hunk 3]
```diff
@@ -28,4 +28,11 @@ export class ServerService {
...
     return this.utils.get('https://api.github.com/repos/infinispan/infinispan/releases/latest', (data) => data.name);
   }
+
+  /**
+   * Get connected clients
+   */
+  public async getConnectedClients(): Promise<Either<ActionResponse, ConnectedClients[]>> {
+    return this.utils.get(this.endpoint + '/connections', (data) => data);
```
Comment (karesti) [inline src/services/serverService.ts:36]: filter duplicates here please

**Review by karesti — CHANGES_REQUESTED** [review]

File: src/services/serverService.ts (hunk 2 above)
Comment (dpanshug) [reply src/services/serverService.ts:28]: Removed in #377

File: src/app/utils/formatAge.ts (hunk 1 above)
Comment (dpanshug) [reply src/app/utils/formatAge.ts:7]: Can't find a library that directly does this formatting. Also added a unit test for this.

**Pull request merged**